package/MDAnalysis/topology/PDBParser.py

```diff
@@ -198,7 +202,9 @@ def parse(self, **kwargs):
             warnings.warn("Invalid atom serials were present, "
                           "bonds will not be parsed")
         else:
-            top.add_TopologyAttr(bonds)
+            # Issue 2832: don't append Bonds if there are no bonds
+            if len(bonds) > 0:
```

Comment:
Can just do "if bonds". Usually better to check the bool-ness rather than qualifying length

Reply:
Do `TopologyAttribute` evaluate to `False` when empty?

Reply:
I didn't think it'd work, but apparently they do :) (or at least Bonds does).